### PR TITLE
Stop pushing prow-test image from prow so that other images can be pushed

### PR DESCRIPTION
**Pull request by chaodaiG**
/cc @chizhg

### CODE REVIEW

**Comment by chizhg** [comment]
@googlebot

File: images/push-all.sh [hunk 1]
```diff
@@ -16,14 +16,17 @@
...
 set -e
 
 CUR_DIR="$(realpath $(dirname $0))"
 
 for dir in ${CUR_DIR}/*; do
-    if [[ -d "${dir}" && -f "${dir}/Makefile" ]]; then
+    if [[ -d "${dir}" && -f "${dir}/Makefile" && "${dir}" != "${CUR_DIR}/${EXCLUDED}" ]]; then
         echo "RUNNING: make -C '${dir}' push"
-        make -C "${dir}" push
+        # make -C "${dir}" push
```
Comment (chizhg) [inline images/push-all.sh:29]: Uncomment?

Comment (chizhg) [reply images/push-all.sh:29]: Also removing `--no-cache` seems to be in your another branch.

Comment (chaodaiG) [reply images/push-all.sh:29]: This is embarrassing, will quickly update it

Comment (chaodaiG) [reply images/push-all.sh]: Done

File: images/simple-image.mk [hunk 2]
```diff
@@ -37,7 +37,7 @@ IMG = $(REGISTRY)/$(PROJECT)/test-infra/$(IMAGE_NAME)
 TAG := $(shell date +v%Y%m%d)-$(shell git describe --always --dirty --match '^$$')
 
 build:
-	docker build --no-cache --pull -t $(IMG):$(TAG) -f $(DOCKERFILE) $(DOCKERBUILDARGS) ../..
```
Comment (chizhg) [inline images/simple-image.mk:40]: IIRC, `--no-cache` was added as we want to always update `gcloud` when we build the `prow-tests` image. Can we make an exception to keep this option when building the `prow-tests` image?

Comment (chaodaiG) [reply images/simple-image.mk]: I see. --no-cache doesn't hurt images other than `prow-tests` as far as I can tell. Let me add it back

**Comment by chizhg** [comment]
@googlebot

**Pull request merged**